### PR TITLE
KAFKA-17121: junit-platform.properties files in published artifacts pollute the test classpath of consumers

### PR DESCRIPTION
**Pull request by xijiu**
Exclude the junit-platform.properties file from `testJar` task

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by chia7712** [comment]
@xijiu thanks for this quick fix. Could you please check the test jar manually?

**Comment by xijiu** [comment]
> @xijiu thanks for this quick fix. Could you please check the test jar manually?

Sure, I will check it

**Comment by xijiu** [comment]
Run the `testJar` task with current trunk branch, the jar contains junit-platform.properties file
image

Add code `exclude 'junit-platform.properties'` and rerun  it, the junit-platform.properties file has disappeared
image

File: build.gradle [hunk 1]
```diff
@@ -588,6 +588,7 @@ subprojects {
       from "$rootDir/LICENSE"
       from "$rootDir/NOTICE"
       from sourceSets.test.output
+      exclude 'junit-platform.properties'
```
Comment (chia7712) [inline build.gradle:594]: Could you please add comments for it?

**Review by chia7712 — COMMENTED** [review]
@xijiu thanks for the verification!

File: build.gradle (hunk 1 above)
Comment (xijiu) [reply build.gradle:594]: Thank you for the reminder, I added the corresponding comment, please review again

Comment (chia7712) [reply build.gradle:594]: We don't need to add jira, and the comments should explain "why" rather that "what it does"

**Comment by wilkinsona** [comment]
LGTM, @chia7712.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**